### PR TITLE
Refactor methods on App and Controls

### PR DESCRIPTION
**Pull request by pvinthemix**
- getTeasFromMood now invokes a method within it to filter for caffeine
-the caffeine buttons on controls now gets pulled from an array of the caffeine levels. looks a little bit cleaner instead of having the same line written 4 times

### CODE REVIEW

File: src/Controls.js [hunk 1]
```diff
@@ -26,13 +26,14 @@ export default class Controls extends Component {
   }
 
   render() {
+    let levels = ['none', 'low', 'moderate', 'high']
+    let buttons = levels.map((level) => {
+      return <button onClick={() => this.findCaffeineLevel({ level })}>{level.charAt(0).toUpperCase() + level.slice(1)}</button>
+    }) 
     return (
```
Comment (shannonmoranetz) [inline src/Controls.js]: Levels array is smart. This is a smarter and shorter way to render four buttons onto the page, by mapping them onto the page. Do we need to uppercase character 0 if we just set the strings inside the array to already be capitalized?

File: src/Controls.js [hunk 2]
```diff
@@ -26,13 +26,14 @@ export default class Controls extends Component {
...
       <div className="controls-container">
           <button className="reset-button" onClick={this.props.toggleSplash}>Reset</button>
-          <button onClick={() => this.findCaffeineLevel('none')}>None</button>
-          <button onClick={() => this.findCaffeineLevel('low')}>Low</button>
-          <button onClick={() => this.findCaffeineLevel('moderate')}>Moderate</button>
-          <button onClick={() => this.findCaffeineLevel('high')}>High</button>
+          {buttons}
           <select className="user-reselected-mood" onChange={this.changeUserSelectedMood}>
```
Comment (shannonmoranetz) [inline src/Controls.js]: Map takes care of this for us and saves a few lines of rendering each button. Perfect.

File: src/App.js [hunk 3]
```diff
@@ -74,23 +74,22 @@ getTeasFromMood() {
...
+    return this.filterTeasByCaffeine(userSelectedTeas);
+  }
+}
+
+filterTeasByCaffeine(userSelectedTeas) {
+  if (this.state.caffeineLevel !== '') {
+    userSelectedTeas = userSelectedTeas.filter((tea) => {
+      return tea.caffeine === this.state.caffeineLevel;
+    })
   }
```
[the remark this answers is not in the document]
Comment (pvinthemix) [reply src/App.js]: No unintended consequences. Functionality worked as normal. This adheres more to SRP.

File: src/Controls.js (hunk 1 above)
Comment (pvinthemix) [reply src/Controls.js]: Nice catch. I Can change that and push it up

File: src/App.js (hunk 3 above)
Comment (Inzurriaga) [inline src/App.js]: great job in taking out the nested if statements and making the app more modular.

File: src/Controls.js (hunk 2 above)
Comment (Inzurriaga) [inline src/Controls.js]: great job on drying up the code!!

**Pull request merged**